packages/migrations/src/actions/2025.03.26T00-00-00.graphql-eslint.v4.ts

```diff
@@ -0,0 +1,104 @@
+import type { MigrationExecutor } from '../pg-migrator';
+
+export default {
+  name: '2025.03.26T00-00-00.graphql-eslint.v4.ts',
+  noTransaction: true,
+  run: async ({ sql, connection }) => {
+    const existingV3Configs = await connection.query(
+      sql`SELECT resource_type, resource_id, config FROM schema_policy_config`,
+    );
+
+    await Promise.all(
+      existingV3Configs.rows.map(async config => {
+        const { resource_type, resource_id, config: v3Config } = config;
+
+        if (v3Config && typeof v3Config === 'object') {
+          const v4Config = migrateConfig(v3Config as any as Record<string, RuleStruct>);
+
+          await connection.query(
+            sql`UPDATE schema_policy_config SET config = ${sql.json(v4Config)} WHERE resource_type = ${resource_type} AND resource_id = ${resource_id}`,
+          );
+          return { resource_type, resource_id, config: v4Config };
+        }
+
+        return null;
+      }),
+    );
```

Comment:
I don't like the idea of starting an uncontrolled amount of database queries in parallel.

Can we please buffer the `schema_policy_config`?

Reply:
Yes, working on it. Fully agree

Reply:
Thanks, refactored it to use Zod and make actual validations on the data. Also it's running it serially now.